mllib/src/main/scala/org/apache/spark/ml/tree/impl/NodeIdCache.scala

```diff
@@ -166,9 +166,13 @@ private[spark] class NodeIdCache(
         }
       }
     }
+    if (nodeIdsForInstances != null) {
+      // Unpersist current one if one exists.
+      nodeIdsForInstances.unpersist(false)
+    }
     if (prevNodeIdsForInstances != null) {
       // Unpersist the previous one if one exists.
-      prevNodeIdsForInstances.unpersist()
+      prevNodeIdsForInstances.unpersist(false)
```

Comment:
Same question as above. `deleteAllCheckpoints` is blocking because it involves calls to `FileSystem.delete`. So, does it make sense to make the call to `unpersist` non-blocking? Am I missing something here?

Reply:
For now `deleteAllCheckpoints` is only called once in whole MLLIB, and current `unpsersit` of `prevNodeIdsForInstances` is in it. So I think we do not need to impl another method to unpersist datasets (like `PeriodicCheckpointer`)